File: test_api.py
import requests
import json

BASE = "http://localhost:8000"

# Test register
print("=== Register ===")
r = requests.post(f"{BASE}/auth/register", json={"email": "test2@test.com", "password": "password123"})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
token = r.json().get("access_token", "")
print()

# Test login
print("=== Login ===")
r = requests.post(f"{BASE}/auth/login", json={"email": "test2@test.com", "password": "password123"})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
token = r.json().get("access_token", "")
print()

headers = {"Authorization": f"Bearer {token}"}

# Test /auth/me
print("=== Me ===")
r = requests.get(f"{BASE}/auth/me", headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test start entry
print("=== Start entry ===")
r = requests.post(f"{BASE}/entries/start", json={"label": "Working on project"}, headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
entry_id = r.json().get("id")
print()

# Test active entry
print("=== Active entry ===")
r = requests.get(f"{BASE}/entries/active", headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test stop entry
print("=== Stop entry ===")
r = requests.post(f"{BASE}/entries/stop", headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test get entries by date
print("=== Entries today ===")
from datetime import date
today = date.today().isoformat()
r = requests.get(f"{BASE}/entries?date={today}", headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test stats
print("=== Stats ===")
r = requests.get(f"{BASE}/stats?from={today}&to={today}", headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test 401 - invalid token
print("=== 401 test (invalid token) ===")
r = requests.get(f"{BASE}/auth/me", headers={"Authorization": "Bearer invalidtoken"})
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test 404 - non-existent entry
print("=== 404 test ===")
r = requests.patch(f"{BASE}/entries/99999", json={"label": "test"}, headers=headers)
print(f"Status: {r.status_code}")
print(f"Response: {r.json()}")
print()

# Test delete
print("=== Delete entry ===")
r = requests.delete(f"{BASE}/entries/{entry_id}", headers=headers)
print(f"Status: {r.status_code}")
print()

# Test start then start again (auto-close)
print("=== Start task 1 ===")
r = requests.post(f"{BASE}/entries/start", json={"label": "Task 1"}, headers=headers)
print(f"Status: {r.status_code}, label: {r.json().get('label')}")
print()

print("=== Start task 2 (should auto-close task 1) ===")
r = requests.post(f"{BASE}/entries/start", json={"label": "Task 2"}, headers=headers)
print(f"Status: {r.status_code}, label: {r.json().get('label')}")
# Check that task 1 is now closed
r2 = requests.get(f"{BASE}/entries/active", headers=headers)
print(f"Active task: {r2.json().get('label')}")
print()

# Stop
requests.post(f"{BASE}/entries/stop", headers=headers)

print("All tests passed!")

File: test_api.ts
const BASE = "http://localhost:8000";

async function main() {
  const email = `test2-${Date.now()}@test.com`;
  const password = "password123";

  // Register
  console.log("=== Register ===");
  let r = await fetch(`${BASE}/auth/register`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password }),
  });
  let data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  let token = data.access_token || "";
  console.log();

  // Login
  console.log("=== Login ===");
  r = await fetch(`${BASE}/auth/login`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ email, password }),
  });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  token = data.access_token || "";
  console.log();

  const headers = { "Authorization": `Bearer ${token}` };

  // /auth/me
  console.log("=== Me ===");
  r = await fetch(`${BASE}/auth/me`, { headers });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // Start entry
  console.log("=== Start entry ===");
  r = await fetch(`${BASE}/entries/start`, {
    method: "POST",
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify({ label: "Working on project" }),
  });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  const entryId = data.id;
  console.log();

  // Active entry
  console.log("=== Active entry ===");
  r = await fetch(`${BASE}/entries/active`, { headers });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // Stop entry
  console.log("=== Stop entry ===");
  r = await fetch(`${BASE}/entries/stop`, { method: "POST", headers });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // Entries today
  console.log("=== Entries today ===");
  const today = new Date().toISOString().slice(0, 10);
  r = await fetch(`${BASE}/entries?date=${today}`, { headers });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // Stats
  console.log("=== Stats ===");
  r = await fetch(`${BASE}/stats?from=${today}&to=${today}`, { headers });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // 401 test
  console.log("=== 401 test (invalid token) ===");
  r = await fetch(`${BASE}/auth/me`, { headers: { "Authorization": "Bearer invalidtoken" } });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // 404 test
  console.log("=== 404 test ===");
  r = await fetch(`${BASE}/entries/99999`, {
    method: "PATCH",
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify({ label: "test" }),
  });
  data = await r.json();
  console.log(`Status: ${r.status}`);
  console.log(`Response: ${JSON.stringify(data)}`);
  console.log();

  // Delete entry
  console.log("=== Delete entry ===");
  r = await fetch(`${BASE}/entries/${entryId}`, { method: "DELETE", headers });
  console.log(`Status: ${r.status}`);
  console.log();

  // Start then start again (auto-close)
  console.log("=== Start task 1 ===");
  r = await fetch(`${BASE}/entries/start`, {
    method: "POST",
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify({ label: "Task 1" }),
  });
  data = await r.json();
  console.log(`Status: ${r.status}, label: ${data.label}`);
  console.log();

  console.log("=== Start task 2 (should auto-close task 1) ===");
  r = await fetch(`${BASE}/entries/start`, {
    method: "POST",
    headers: { ...headers, "Content-Type": "application/json" },
    body: JSON.stringify({ label: "Task 2" }),
  });
  data = await r.json();
  console.log(`Status: ${r.status}, label: ${data.label}`);
  r = await fetch(`${BASE}/entries/active`, { headers });
  const activeData = await r.json();
  console.log(`Active task: ${activeData.label}`);
  console.log();

  // Stop
  await fetch(`${BASE}/entries/stop`, { method: "POST", headers });

  console.log("All tests passed!");
}

main().catch(console.error);
